Local File Trigger on 1.107.0 fails with "Cannot find module 'chokidar'"

### Bug Description

Workflows including "Local File Trigger" nodes, are impossible to run or activate, as said node crashes with `"Cannot find module 'chokidar'"` error:

```
Aug 12 14:40:37 deedee docker-n8n-start[1104098]: Activation of workflow "Local Test" (tq2GnSM5qjhVjoVK) did fail with error: "Cannot find module 'chokidar'

Aug 12 14:40:37 deedee docker-n8n-start[1104098]: Require stack:
Aug 12 14:40:37 deedee docker-n8n-start[1104098]: - /usr/local/lib/node_modules/n8n/node_modules/.pnpm/n8n-nodes-base@file+packages+nodes-base_@aws-sdk+credential-providers@3.808.0_asn1.js@5_1af219c3f47f2a1223ec4ccec249a974/node_modules/n8n-nodes-base/dist/nodes/LocalFileTrigger/LocalFileTrigger.node.js
Aug 12 14:40:37 deedee docker-n8n-start[1104098]: - /usr/local/lib/node_modules/n8n/node_modules/.pnpm/n8n-core@file+packages+core_@opentelemetry+api@1.9.0_@opentelemetry+sdk-trace-base@1.30_0c275070fd0434c00ffd62213b2bcae4/node_modules/n8n-core/dist/nodes-loader/load-class-in-isolation.js
Aug 12 14:40:37 deedee docker-n8n-start[1104098]: - /usr/local/lib/node_modules/n8n/node_modules/.pnpm/n8n-core@file+packages+core_@opentelemetry+api@1.9.0_@opentelemetry+sdk-trace-base@1.30_0c275070fd0434c00ffd62213b2bcae4/node_modules/n8n-core/dist/nodes-loader/directory-loader.js
Aug 12 14:40:37 deedee docker-n8n-start[1104098]: - /usr/local/lib/node_modules/n8n/node_modules/.pnpm/n8n-core@file+packages+core_@opentelemetry+api@1.9.0_@opentelemetry+sdk-trace-base@1.30_0c275070fd0434c00ffd62213b2bcae4/node_modules/n8n-core/dist/nodes-loader/index.js
Aug 12 14:40:37 deedee docker-n8n-start[1104098]: - /usr/local/lib/node_modules/n8n/node_modules/.pnpm/n8n-core@file+packages+core_@opentelemetry+api@1.9.0_@opentelemetry+sdk-trace-base@1.30_0c275070fd0434c00ffd62213b2bcae4/node_modules/n8n-core/dist/index.js
Aug 12 14:40:37 deedee docker-n8n-start[1104098]: - /usr/local/lib/node_modules/n8n/node_modules/.pnpm/@n8n+db@file+packages+@n8n+db_@opentelemetry+api@1.9.0_@opentelemetry+sdk-trace-base@1._09eeae95866204ce68085ead160befa7/node_modules/@n8n/db/dist/repositories/execution.repository.js
Aug 12 14:40:37 deedee docker-n8n-start[1104098]: - /usr/local/lib/node_modules/n8n/node_modules/.pnpm/@n8n+db@file+packages+@n8n+db_@opentelemetry+api@1.9.0_@opentelemetry+sdk-trace-base@1._09eeae95866204ce68085ead160befa7/node_modules/@n8n/db/dist/repositories/index.js
Aug 12 14:40:37 deedee docker-n8n-start[1104098]: - /usr/local/lib/node_modules/n8n/node_modules/.pnpm/@n8n+db@file+packages+@n8n+db_@opentelemetry+api@1.9.0_@opentelemetry+sdk-trace-base@1._09eeae95866204ce68085ead160befa7/node_modules/@n8n/db/dist/index.js
Aug 12 14:40:37 deedee docker-n8n-start[1104098]: - /usr/local/lib/node_modules/n8n/dist/commands/start.js
Aug 12 14:40:37 deedee docker-n8n-start[1104098]: - /usr/local/lib/node_modules/n8n/dist/command-registry.js" | retry in 512 seconds
```

I assume it may be related to https://github.com/n8n-io/n8n/pull/18094 which removed chokidar from dependencies.

### To Reproduce

1. Be on `1.107.0` version
2. Create empty workflow
3. Add "Local File Trigger" node (doesn't have to be configured even)
4. Save workflow
5. Try to activate workflow

### Expected behavior

Workflows with "Local File Trigger" nodes can be activated and work as expected.

### Debug Info

-

### Operating System

NixOS 25.11 (but n8n is running in official docker container)

### n8n Version

1.107.0

### Node.js Version

v22.17.0

### Database

SQLite (default)

### Execution mode

main (default)

### Hosting

self hosted

--- a/package.json
+++ b/package.json
@@ -88,7 +88,7 @@
       "@azure/identity": "^4.3.0",
       "@n8n/typeorm>@sentry/node": "catalog:",
       "@types/node": "^20.17.50",
-      "chokidar": "4.0.1",
+      "chokidar": "4.0.3",
       "esbuild": "^0.24.0",
       "multer": "^2.0.2",
       "prebuild-install": "7.1.3",

--- a/package.json
+++ b/package.json
@@ -88,7 +88,7 @@
       "@azure/identity": "^4.3.0",
       "@n8n/typeorm>@sentry/node": "catalog:",
       "@types/node": "^20.17.50",
-      "chokidar": "^4.0.1",
+      "chokidar": "4.0.1",
       "esbuild": "^0.24.0",
       "multer": "^2.0.2",
       "prebuild-install": "7.1.3",

--- a/packages/nodes-base/nodes/LocalFileTrigger/LocalFileTrigger.node.ts
+++ b/packages/nodes-base/nodes/LocalFileTrigger/LocalFileTrigger.node.ts
@@ -1,4 +1,5 @@
 import { watch } from 'chokidar';
+import type { EventName } from 'chokidar/handler';
 import {
 	type ITriggerFunctions,
 	type IDataObject,
@@ -233,11 +234,11 @@ export class LocalFileTrigger implements INodeType {
 		const path = this.getNodeParameter('path') as string;
 		const options = this.getNodeParameter('options', {}) as IDataObject;
 
-		let events: string[];
+		let events: EventName[];
 		if (triggerOn === 'file') {
 			events = ['change'];
 		} else {
-			events = this.getNodeParameter('events', []) as string[];
+			events = this.getNodeParameter('events', []) as EventName[];
 		}
 		const ignored = options.ignored === '' ? undefined : (options.ignored as string);
 		const watcher = watch(path, {
@@ -259,7 +260,7 @@ export class LocalFileTrigger implements INodeType {
 		};
 
 		for (const eventName of events) {
-			watcher.on(eventName, (pathString) => executeTrigger(eventName, pathString as string));
+			watcher.on(eventName, (pathString: string) => executeTrigger(eventName, pathString));
 		}
 
 		async function closeFunction() {

--- a/packages/nodes-base/package.json
+++ b/packages/nodes-base/package.json
@@ -892,6 +892,7 @@
     "basic-auth": "catalog:",
     "change-case": "4.1.2",
     "cheerio": "1.0.0-rc.6",
+    "chokidar": "catalog:",
     "cron": "3.1.7",
     "csv-parse": "5.5.0",
     "currency-codes": "2.1.0",

--- a/pnpm-workspace.yaml
+++ b/pnpm-workspace.yaml
@@ -24,6 +24,7 @@ catalog:
   axios: 1.8.3
   basic-auth: 2.0.1
   callsites: 3.1.0
+  chokidar: 4.0.1
   fast-glob: 3.2.12
   flatted: 3.2.7
   form-data: 4.0.0

--- a/pnpm-workspace.yaml
+++ b/pnpm-workspace.yaml
@@ -24,7 +24,7 @@ catalog:
   axios: 1.8.3
   basic-auth: 2.0.1
   callsites: 3.1.0
-  chokidar: 4.0.1
+  chokidar: 4.0.3
   fast-glob: 3.2.12
   flatted: 3.2.7
   form-data: 4.0.0

